pkg/daemon/ceph/osd/volume.go

```diff
@@ -127,9 +146,29 @@ func (a *OsdAgent) initializeBlockPVC(context *clusterd.Context, devices *Device
 				"--data",
 				deviceArg,
 			}...)
+
 			// execute ceph-volume with the device
+			op, err := context.Executor.ExecuteCommandWithCombinedOutput(false, "", baseCommand, immediateExecuteArgs...)
+			if err != nil {
+				cvLogFilePath := path.Join(cvLogDir, "ceph-volume.log")
+
+				// Open c-v log file
+				cvLogFile, err := os.Open(cvLogFilePath)
+				if err != nil {
+					logger.Errorf("failed to open ceph-volume log file %q. %v", cvLogFilePath, err)
+				}
+				defer cvLogFile.Close()
+
+				// Read c-v log file
+				b, err := ioutil.ReadAll(cvLogFile)
+				if err != nil {
+					logger.Errorf("failed to read ceph-volume log file %q. %v", cvLogFilePath, err)
+				}
+
+				// Print c-v log before exiting
+				logger.Errorf("%s", string(b))
 
-			if op, err := context.Executor.ExecuteCommandWithCombinedOutput(false, "", baseCommand, immediateExecuteArgs...); err != nil {
+				// Return failure
 				return "", errors.Wrapf(err, "failed ceph-volume") // fail return here as validation provided by ceph-volume
```

Comment:
This is not the original err from the c-v command failure, we'll need to use a different var

Reply:
Good catch.